src/components/ShoppingList/ShoppingList.js

```diff
@@ -168,18 +168,34 @@ function ShoppingList({ listId, handleModalOpen }) {
       item.numberOfPurchases,
     );
 
-    db.collection(`lists/${listId}/items`)
+    return db
+      .collection(`lists/${listId}/items`)
       .doc(item.id)
       .update({
-        lastPurchaseDate: new firebase.firestore.Timestamp(
-          newPurchaseDate.toSeconds(),
-          0,
-        ),
+        lastPurchaseDate: firebase.firestore.FieldValue.serverTimestamp(),
         numberOfPurchases: firebase.firestore.FieldValue.increment(1),
         purchaseInterval: newPurchaseInterval,
-      })
-      .catch((err) => {
-        console.log(err);
+        // back up some info in case user mistakenly checks item and wants to undo
+        undoRestore: {
+          lastPurchaseDate: item.lastPurchaseDate,
+          purchaseInterval: item.purchaseInterval,
+        },
+      });
+  };
+
+  // restores item's previous lastPurchaseDate, purchaseInterval and numberOfPurchases
+  // for use when a user accidentally checks an item off, and wants to undo
+  const uncheckAsPurchased = (item) => {
+    return db
+      .collection(`lists/${listId}/items`)
+      .doc(item.id)
+      .update({
+        // decrement total purchases by 1
+        numberOfPurchases: firebase.firestore.FieldValue.increment(-1),
```

Comment:
I was wondering why you didn't use "decrement" then saw on the docs for firestore `FieldValue` that there is no decrement method. I wonder why?

Reply:
Yes, I thought that was kind of quirky as well.